src/parallel/StaticIrregDomainDecomposition.cpp

```diff
@@ -49,17 +49,56 @@ void StaticIrregDomainDecomposition::readXML(XMLfileUnits &xmlconfig) {
   DomainDecompMPIBase::readXML(xmlconfig);
   // bypass DomainDecomposition readXML to avoid reading MPIGridDims
 
-  std::string filename = xmlconfig.getNodeValue_string("subdomainWeightsCSV");
-  if (!filename.empty()) {
-    updateSubdomainWeightsFromFile(filename);
-    for (int i = 0; i < _subdomainWeights.size(); ++i) {
+  if (xmlconfig.changecurrentnode("subdomainWeights")) {
+    /* If the node does not exist, then behaviour is identical to
+     * DomainDecomposition unless the weights are set through the constructor
+     */
+    Log::global_log->info()
+        << "Reading weights for StaticIrregDomainDecomposition" << std::endl;
```

Comment:
This log feels a bit unnecessary. I'd only log what was actually parsed or a meaningful error if parsing failed. But if you like to keep it I'm also fine with it.

Reply:
I guess it's better as a debug, that was the purpose I was using it for as well